wandb/sdk/wandb_run.py

```diff
@@ -908,14 +906,12 @@
     @_run_decorator._attach
     def id(self) -> str:
         """Identifier for this run."""
-        if TYPE_CHECKING:
-            assert self._settings.run_id is not None
-        return self._settings.run_id
+        return self._settings.run_id or ""
```

Comment:
I think assert would be better: if run_id is not set, that sounds like a bug in wandb. The TYPE_CHECKING was unnecessary though.